packages/babel-helpers/src/helpers/applyDecs2305.ts

```diff
@@ -674,32 +189,407 @@ function defineMetadata(Class: any, metadata: any) {
   initializeClass(Class);
  */
 
-export default function applyDecs2305(
-  targetClass: object,
+export default /* @no-mangle */ function applyDecs2305(
+  targetClass: any,
   memberDecs: DecoratorInfo[],
   classDecs: Function[],
   classDecsHaveThis: number,
   instanceBrand: Function,
   parentClass: any,
 ) {
+  function _bindPropCall(obj: any, name: string, before?: Function) {
+    return function (_this: any, value?: any) {
+      if (before) {
+        before(_this);
+      }
+      return obj[name].call(_this, value);
+    };
+  }
+
+  function runInitializers(initializers: Function[], value: any) {
+    for (var i = 0; i < initializers.length; i++) {
+      initializers[i].call(value);
+    }
+    return value;
+  }
+
+  function assertCallable(
+    fn: any,
+    hint1: string,
+    hint2?: string,
+    throwUndefined?: boolean,
+  ) {
+    if (typeof fn !== "function") {
+      if (throwUndefined || fn !== void 0) {
+        throw new TypeError(
+          hint1 +
+            " must " +
+            (hint2 || "be") +
+            " a function" +
+            (throwUndefined ? "" : " or void 0"),
```

Comment:
```suggestion
            (throwUndefined ? "" : " or undefined"),
```

Most people never see `void 0` when writing JS code :)